src/varity/ref_gene.clj

```diff
@@ -208,28 +208,55 @@
            (map (partial read-exon-sequence seq-rdr))
            string/join))
 
+(defn exon-ranges->intron-ranges
+  [exon-ranges]
+  (->> exon-ranges
+       ((fn [r]
+          (loop [intron-ranges []
+                 left (first r)
+                 right (second r)
+                 ranges (rest r)]
+            (if (empty? (first ranges)) intron-ranges
+                (recur (conj intron-ranges [(inc (second left)) (dec (first right))])
+                       right
+                       (second ranges)
+                       (rest ranges))))))))
```

Comment:
I think this arrow macro `->>` is not elegant. It's better to describe in simple `(loop [] ... (recure)` way.

Reply:
I'm sorry, you are right. 🙇‍♂️

Reply:
I’m not a designated reviewer, but let me add a few comments on this.

You can rewrite the code more simply using [destructing](https://clojure.org/guides/destructuring):

```clojure
(defn exon-ranges->intron-ranges
  [exon-ranges]
  (loop [intron-ranges []
         [left & [right :as ranges]] exon-ranges]
    (if (empty? right)
      intron-ranges
      (recur (conj intron-ranges [(inc (second left)) (dec (first right))])
             ranges))))
```

Or if you don’t mind additional allocation here, `partiton` will make it even shorter:

```clojure
(defn exon-ranges->intron-ranges
  [exon-ranges]
  (mapv (fn [[left right]] [(inc (second left)) (dec (first right))])
        (partition 2 1 exon-ranges)))
```

The more code you write, the more likely it is to have a bug in it, so it’s good to use existing powerful features effectively and keep our code simple as far as we can.

Reply:
Thank you for your review.

I have learned about `partition` function.
The code using `partition` looks more elegant than old one.

I'd like to write code with simplicity in mind ...